OutOfSchool/OutOfSchool.WebApi/Services/WorkshopServicesCombiner.cs

```diff
@@ -171,13 +207,27 @@ public async Task<List<WorkshopCard>> GetByProviderId(Guid id)
             return workshopCards.ToList();
         }
 
-        // <inheritdoc/>
+        /// <inheritdoc/>
         public async Task<Guid> GetWorkshopProviderId(Guid workshopId) =>
             await workshopService.GetWorkshopProviderOwnerIdAsync(workshopId).ConfigureAwait(false);
 
-        private List<WorkshopCard> DtoModelsToWorkshopCards(IEnumerable<WorkshopDTO> source)
+        public async Task<IEnumerable<string>> GetNotificationsRecipientIds(NotificationAction action, Dictionary<string, string> additionalData, Guid objectId)
         {
-            return source.Select(currentElement => currentElement.ToCardDto()).ToList();
+            var recipientIds = new List<string>();
+
+            Expression<Func<Favorite, bool>> filter = x => x.WorkshopId == objectId;
+            var favoriteWorkshopUsersIds = favoriteRepository.Get(where: filter).Select(x => x.UserId).ToList();
+
+            Expression<Func<Application, bool>> predicate =
+                x => x.Status != ApplicationStatus.Left &&
+                     x.WorkshopId == objectId;
+            var applications = await applicationRepository.GetByFilter(predicate).ConfigureAwait(false);
+            var appliedUsersIds = applications.Select(x => x.Parent.UserId);
```

Comment:
Performance-wise it is better to use Select on IQueryable, i.e. before you make call to the DB
```suggestion
            var appliedUsersIds = await applicationRepository.Get(where: predicate)
                .Select(x => x.Parent.UserId)
                .ToListAsync()
                .ConfigureAwait(false);
```